bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/bridge/BoschHttpClient.java

```diff
@@ -249,13 +252,13 @@ public boolean doPairing() throws InterruptedException {
                 logger.info("Pairing failed with response status {}.", contentResponse.getStatus());
                 return false;
             }
-        } catch (TimeoutException | CertificateEncodingException | KeyStoreException | NullPointerException e) {
-            logger.warn("Pairing failed with exception {}", e.getMessage());
+        } catch (TimeoutException | CertificateEncodingException | KeyStoreException | RuntimeException e) {
+            logger.warn("Pairing failed with exception {}", e.getMessage(), e);
             return false;
         } catch (ExecutionException e) {
             // javax.net.ssl.SSLHandshakeException: General SSLEngine problem
             // => usually the pairing failed, because hardware button was not pressed.
-            logger.trace("Pairing failed - Details: {}", e.getMessage());
+            logger.trace("Pairing failed - Details: {}", e.getMessage(), e);
```

Comment:
As far as I'm able remember one of first contribution to the binding, this ExecutionException can and will happen.
Compare also with comment. 
Therefore this is not a bug which can be fixed and should not log the stack trace from my point of view. Compare with [CodeGuideline](https://www.openhab.org/docs/developer/guidelines.html#f-logging) says "Configuration errors by users should only print log messages about what's wrong."